lib/context_adapter_helper.js

```diff
@@ -333,12 +408,13 @@ function getOperationDescriptor(request, callback) {
 /**
  * Checks that an queryContext response returned by the Context Adapter to get
  *  service information is well-formed
+ * @param {object} context The context for logging
  * @param {object} response The response received from the Context Adapter
  * @param {string} serviceId The identifier of the service whose information was retrieved
  * @param {function} callback Callback to be called with the result of the check
  * @return {*} True if the queryContextResponse is valid. False otherwise.
  */
-function checkQueryContextResponse(response, serviceId, callback) {
+function checkQueryContextResponse(context, response, serviceId, callback) {
```

Comment:
The context depends on each request for better traceability (see https://github.com/telefonicaid/context-adapter/blob/task/synchronization-after-happy-path-e2e-integration/lib/context_adapter_server.js#L241) In fact, it depends on the UNICA_CORRELATOR header if present on the original request or a new one is generated for each specific request :)

I could not do it this way if you think that approach is the right one and obviously the context would be static at a JS file level with no need to pass it :)

Another approach to try to keep all the benefits and your suggestion would be to include the context in the response object passed via:
```javascript
response.contextAdapter.context = theConcreteContextBasedOnEachRequest;
```
and I could take it from there. What do you think?